### PR TITLE
WKB encoder follow up

### PR DESCRIPTION
**Pull request by kylebarron**
- Wrap `options` in `wkb` key.
- More unified test cases for loader and writer. Use `encodeSync` in tests instead of calling `encodeWKB` directly.
- Docs page.
- What's new

### CODE REVIEW

File: modules/wkt/docs/api-reference/wkb-writer.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const geometry = {
+  type: "Polygon",
+  coordinates: [[[1, 2], [3, 4], [5, 6], [1, 2]]]
+}
+const arrayBuffer = encodeSync(geometry, WKBWriter, {wkt: {hasZ: false, hasM: false}})
+```
+
+## Options
+
+- `hasZ`: Should be `true` if the GeoJSON input has Z values. These values are expected to be the third coordinate position.
```
Comment (ibgreen) [inline modules/wkt/docs/api-reference/wkb-writer.md:36]: As I recall, these were not really options in WKX but fields on their base geometry class. I turned them into options as the simplest way forward..

Do we normally autodetect these for other formats? Should we try to do that longer term also for WKB?

File: modules/wkt/docs/api-reference/wkb-writer.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+It's essentially a binary representation of WKT. For common geospatial types
+including (Multi) `Point`, `Line`, and `Polygon`, there's a 1:1 correspondence
+between WKT/WKB and GeoJSON. WKT and WKB also support extended geometry types,
+such as `Curve`, `Surface`, and `TIN`, which don't have a correspondence to
+GeoJSON.
+
+- Coordinates can be 2-4 dimensions and are interleaved.
+- Positions stored as double precision
+
+![image](https://user-images.githubusercontent.com/15164633/83707157-90413b80-a5d6-11ea-921c-b04208942e79.png)
```
Comment (ibgreen) [inline modules/wkt/docs/api-reference/wkb-writer.md:57]: An short attribution section mentioning WKX here (and a longer one in the main page for the module) would be nice.

File: modules/wkt/docs/api-reference/wkb-writer.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+const arrayBuffer = encodeSync(geometry, WKBWriter, {wkt: {hasZ: false, hasM: false}})
+```
+
+## Options
+
+- `hasZ`: Should be `true` if the GeoJSON input has Z values. These values are expected to be the third coordinate position.
+- `hasM`: Should be `true` if the GeoJSON input has M values. Thes are expected to be the third coordinate position if Z values do not exist, or fourth if Z values do exist.
+
+## Format Summary
```
Comment (ibgreen) [inline modules/wkt/docs/api-reference/wkb-writer.md:39]: Nice writeup. Do we duplicate this between the writer and loader pages or consolidate on one of them or the main page for the module?

File: modules/wkt/src/lib/encode-wkb.ts [hunk 4]
```diff
@@ -46,12 +51,16 @@ interface WKBOptions {
...
 export default function encodeWKB(
   geometry: Geometry | Feature,
-  options: WKBOptions = {hasZ: false, hasM: false}
+  options: WKBOptions | {wkt: WKBOptions}
 ): ArrayBuffer {
   if (geometry.type === 'Feature') {
     geometry = geometry.geometry;
   }
 
+  if ('wkt' in options) {
```
Comment (ibgreen) [inline modules/wkt/src/lib/encode-wkb.ts:60]: I am not sure that we support nested options in the encoders. The Writer/encode... system is not nearly as built out as the Loader/parse... system.

File: modules/wkt/src/wkb-writer.ts [hunk 5]
```diff
@@ -12,8 +12,11 @@ export const WKBWriter: Writer = {
   version: VERSION,
   extensions: ['wkb'],
   // @ts-ignore
-  encode: encodeWKB,
+  encodeSync: encodeWKB,
   options: {
-    wkt: {}
+    wkt: {
```
Comment (ibgreen) [inline modules/wkt/src/wkb-writer.ts:17]: Not even sure these are applied, but doesn't hurt writing them out,.

**Review by ibgreen — APPROVED** [review]
Looks great, thanks!

File: modules/wkt/docs/api-reference/wkb-writer.md (hunk 3 above)
Comment (kylebarron) [reply modules/wkt/docs/api-reference/wkb-writer.md:39]: This was copied from the loader page

File: modules/wkt/src/lib/encode-wkb.ts (hunk 4 above)
Comment (kylebarron) [reply modules/wkt/src/lib/encode-wkb.ts:60]: This should support both nested and non-nested options as-is

File: modules/wkt/docs/api-reference/wkb-writer.md (hunk 1 above)
Comment (kylebarron) [reply modules/wkt/docs/api-reference/wkb-writer.md:36]: WKB has different concepts for Z and M (measure) coordinates. By having these as options, if your data has 3 dimensions you could declare the 3rd dimension to be and M coordinate, not a Z coordinate. Shapefile I believe also has this distinction, but when parsing to GeoJSON we don't record what dimension the 3rd coordinate value is

**Pull request merged**